openmc/deplete/microxs.py

```diff
@@ -20,168 +22,176 @@
 _valid_rxns.append('fission')
 
 
-class MicroXS(DataFrame):
-    """Microscopic cross section data for use in transport-independent depletion.
-
-    .. versionadded:: 0.13.1
+def get_microxs_and_flux(
+        model: openmc.Model,
+        domains,
```

Comment:
can we make this optional and default to all materials or all cells?

Reply:
The way it's currently designed, this might be a bit confusing because you get back lists of fluxes and micros corresponding to the order that was used in `domains`. If we default to something, the user won't have any idea of the order.

Reply:
Should we instead return a `dict` with material, cell, or universe id as the key and a tuple of `(MicroXS, flux)` as the value? In all likelihood the order of `domains` means nothing in particular so I'd be in favor of making order not important at all.

Reply:
I guess knowing the cell id might be useful if we want to remove particular cell from the geometry for an R2S simulation with moving geometry.

Reply:
Sorry for the late follow up -- @eepeterson and I talked offline about this but just wanted to summarize the discussion. The reason I've chosen to return a list rather than a dict is mostly related to memory. If you have a long list of domains in a model that you need cross sections for, returning a dict would require more memory whereas a list keeps it as small as possible. If a user really wants a dictionary, it would easy enough for them to create one in a one-liner.